src/shared_modules/utils/socketWrapper.hpp

```diff
@@ -110,6 +110,163 @@ class TcpAddress final : public SockAddress<TcpAddress>
     }
 };
 
+/**
+ * @brief This class is used for the following format:
+ *          - 4 bytes: packet size. Mandatory size of the whole packet (body, optional header and header size)
+ *          - 4 bytes: header size. Mandatory size of the optional header. It can be 0.
+ *          - N bytes: header. Optional header.
+ *          - M bytes: body. Mandatory data to send.
+ */
+class appendHeaderProtocol final
+{
+public:
+    /**
+     * @brief Create the buffer to send according to this protocol.
+     *
+     * @param buffer        Output buffer to write.
+     * @param bufferSize    Size of the output buffer.
+     * @param dataBody      Data to send.
+     * @param sizeBody      Size of the data to send.
+     * @param dataHeader    Optional header to send.
+     * @param sizeHeader    Size of the optional header.
+     */
+    void static buildBuffer(std::vector<char>& buffer,
+                            uint32_t& bufferSize,
+                            const char* dataBody,
+                            uint32_t sizeBody,
+                            const char* dataHeader = nullptr,
+                            uint32_t sizeHeader = 0)
+    {
+        if (PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody > BUFFER_SIZE)
+        {
+            buffer.resize(PACKET_SIZE + HEADER_SIZE + sizeHeader + sizeBody + 1);
+        }
+
+        // Write packet size to the buffer.
+        uint32_t* uip;
+        uip = (uint32_t*)buffer.data();
+        *uip = sizeBody + HEADER_SIZE + sizeHeader;
+
+        // Write header size to the buffer.
+        uip = (uint32_t*)(buffer.data() + PACKET_SIZE);
```

Comment:
I suggest defining a packaged struct with all the protocol header content and then using a pointer to that type of struct for referencing different fields in it. Maybe in the future, if we add some extra field to the protocol header, then a struct approach could be beneficial

Reply:
Instead of doing:
```
uint32_t* uip;
uip = (uint32_t*)buffer.data();
*uip = sizeBody + HEADER_SIZE + sizeHeader;
uip = (uint32_t*)(buffer.data() + PACKET_SIZE);
*uip = sizeHeader;
```
I suggest defining a struct for the header (let's call it struct Header for this example):
```
struct Header * pHeader = static_cast<struct Header *>(buffer.data());
pHeader->packetSize = sizeBody + HEADER_SIZE + sizeHeader;
pHeader->headerSize = sizeHeader;
```

Reply:
Ok, thank you.